lib/clientv2/ippool.go

```diff
@@ -88,6 +96,46 @@ func (r ipPools) Update(ctx context.Context, res *apiv2.IPPool, opts options.Set
 
 // Delete takes name of the IPPool and deletes it. Returns an error if one occurs.
 func (r ipPools) Delete(ctx context.Context, name string, opts options.DeleteOptions) (*apiv2.IPPool, error) {
+	// Deleting a pool requires a little care because of existing endpoints
+	// using IP addresses allocated in the pool.  We do the deletion in
+	// the following steps:
+	// -  disable the pool so no more IPs are assigned from it
+	// -  remove all affinities associated with the pool
+	// -  delete the pool
+	logCxt := log.WithField("Name", name)
+
+	// Get the pool so that we can find the CIDR associated with it.  If the pool is
+	// enabled we'll need to release affinities associated with the pool.
+	pool, err := r.Get(ctx, name, options.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	logCxt = logCxt.WithField("CIDR", pool.Spec.CIDR)
+
+	// If this pool is enabled then we'll need to release affinities associated with this
+	// pool.
+	if !pool.Spec.Disabled {
+		// Set the disabled flag to ensure we stop allocating from this pool.
+		logCxt.Debugf("Disabling pool to release affinities")
+		pool.Spec.Disabled = true
+		if _, err := r.Update(ctx, pool, options.SetOptions{}); err != nil {
+			return nil, err
+		}
+
+		// Convert the CIDR to a network.  This shouldn't fail, but if it does then there
+		// should be no need to release affinities since we can't assigning from this invalid
+		// pool.
+		if _, cidrNet, err := cnet.ParseCIDR(pool.Spec.CIDR); err != nil {
+			logCxt.Debugf("Releasing pool affinities")
+			err = r.client.IPAM().ReleasePoolAffinities(context.Background(), *cidrNet)
```

Comment:
Shouldn't we pass in `ctx` instead of a new context?

Reply:
ha!  yeah absolutely -- copied from elsewhere and didn't spot that.